src/amoc_telemetry.erl

```diff
@@ -2,14 +2,24 @@
 %% @copyright 2023 Erlang Solutions Ltd.
 -module(amoc_telemetry).
 
--export([execute/3]).
+-export([execute/3, execute_log/4]).
 
 -spec execute(EventName, Measurements, Metadata) -> ok when
       EventName :: telemetry:event_name(),
       Measurements :: telemetry:event_measurements(),
       Metadata :: telemetry:event_metadata().
 execute(Name, Measurements, Metadata) ->
     TimeStamp = erlang:monotonic_time(),
-    NameWithAmocPrefix = [amoc | Name],
+    PrefixedName = [amoc | Name],
     MetadataWithTS = Metadata#{monotonic_time => TimeStamp},
-    telemetry:execute(NameWithAmocPrefix, Measurements, MetadataWithTS).
+    telemetry:execute(PrefixedName, Measurements, MetadataWithTS).
+
+-spec execute_log(Level, EventName, Metadata, Msg) -> ok when
+      Level :: logger:level(),
+      EventName :: telemetry:event_name(),
+      Metadata :: telemetry:event_metadata(),
+      Msg :: binary().
+execute_log(Level, Name, Metadata, Message) ->
+    PrefixedName = [amoc | Name],
+    MetadataWithLog = Metadata#{log_level => Level, msg => Message},
+    telemetry:execute(PrefixedName, #{Level => 1}, MetadataWithLog).
```

Comment:
why not reusing amoc_telemetry:execute/3 here? don't we want `monotonic_time` in metadata?

anyway, this is minor thing. can be added later.